frontend/hooks/save-settings/index.tsx

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useState } from 'react'
+import { useJsonRpcClient } from '../../contexts/json-rpc/json-rpc-context'
+import { RpcMethods } from '../../lib/client-rpc-methods'
+import { useGlobalsStore } from '../../stores/globals'
+
+export const useSaveSettings = () => {
+  const [loading, setLoading] = useState(false)
+  const { request } = useJsonRpcClient()
+  const { loadGlobals } = useGlobalsStore((state) => ({
+    loadGlobals: state.loadGlobals
+  }))
+  const save = useCallback(
```

Comment:
**chore (non-blocking)**:  I'd have appreciated some sort of comment on this one, but it's not essential. It probably does provide a hook that persists settings. I don't understand why it needs to `loadGlobals` after saving, but... maybe that's fine?